src/core/mac/mac.cpp

```diff
@@ -165,6 +165,11 @@ Mac::Mac(ThreadNetif &aThreadNetif):
     mTxFrame = static_cast<Frame *>(otPlatRadioGetTransmitBuffer(mNetif.GetInstance()));
 }
 
+void Mac::PrepareRadio()
+{
+    otPlatRadioSetCallbacks(mNetif.GetInstance(), ReceiveDoneTask, TransmitDoneTask);
```

Comment:
It has to be set dynamically because NCP previously could have set the callbacks. For instance, say the Host first wanted to use the radio layer directly, using the `SPINEL_PROP_STREAM_RAW` to write MAC frames directly. Then it wanted to switch to use Thread, calling `_NET_UP` and `_THREAD_START`. That calls `otInterfaceUp` which then calls into `Mac::PrepareRadio` eventually, resetting the callbacks.